vxlan/vxlan_utils.go

```diff
@@ -0,0 +1,114 @@
+package vxlan
+
+import (
+	"net"
+	"syscall"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	linkUpRetries = 3
+)
+
+type vxlanIntfInfo struct {
+	name string
+	vni  int
+	port int
+	mac  net.HardwareAddr
+	mtu  int
+}
+
+func createVxlanInterface(v *vxlanIntfInfo) error {
+	logrus.Debugf("vxlan: creating interface %+v", v)
+
+	vxlan := &netlink.Vxlan{
+		LinkAttrs: netlink.LinkAttrs{Name: v.name, MTU: v.mtu},
+		VxlanId:   v.vni,
+		Learning:  false,
+		Port:      v.port,
+		Proxy:     true,
+		L3miss:    true,
+		L2miss:    true,
+		RSC:       true,
+	}
+
+	err := netlink.LinkAdd(vxlan)
+	if err != nil && err != syscall.EEXIST {
+		return err
+	}
+
+	err = netlink.LinkSetHardwareAddr(vxlan, v.mac)
+	if err != nil {
+		return err
+	}
+
+	linkUpSuccessful := false
+	retries := linkUpRetries
+	for retries > 0 {
```

Comment:
I don‘t know why.
These code copied from rancher-net.
I also want to know what was the history behind this.